### DISCUSSION TITLE
CRON job won't execute when using the anon/service_role

### DISCUSSION THREAD

**Discussion by benj-webriq** [supabase/supabase#31187] [Questions] [answered]
Hi, I've already set the permission for both roles to be able to manage the cron schema and its tables but everytime I spawn a new job. the job won't execute but it's a different case when using the postgres role in the table editor inside supabase.

**Comment by GaryAustin1** [comment, marked as answer]
You might try using a security definer function to make the calls.

Here is another discussion on setting grants...
https://github.com/orgs/supabase/discussions/27917

**Reply by benj-webriq** [reply]
I tried granting access to service role and anon, and still doesn't work.
![image](https://github.com/user-attachments/assets/1219e8be-1dc7-4d79-8681-3cd802072d5c)

**Reply by GaryAustin1** [reply]
You said the cron job would not execute.

Your error is http_post failed to connect.
Not sure that has to do with cron and roles.

**Reply by benj-webriq** [reply]
> You said the cron job would not execute.
> 
> Your error is http_post failed to connect. Not sure that has to do with cron and roles.

Hmm, my bad. So the problem here is the pgnet extension then? I'm in the table editor and it says "RLS disabled" might be the reason why it won't connect.

**Reply by benj-webriq** [reply]
Aight, my bad. The net.http_post only allows the postgres role or those roles with super privileges as executor unless I define definer function as you've mentioned. Thanks.

**Comment by benj-webriq** [comment]
Solved! Thanks bro, I'll close this now 🙏
